pallets/creator-staking/src/functions.rs

```diff
@@ -0,0 +1,309 @@
+        space_id: SpaceId,
+        unregister_origin: UnregistrationAuthority<T::AccountId>,
+    ) -> DispatchResultWithPostInfo {
+        let mut creator_info =
+            RegisteredCreators::<T>::get(space_id).ok_or(Error::<T>::NotRegisteredCreator)?;
+
+        ensure!(creator_info.state == CreatorState::Registered, Error::<T>::NotRegisteredCreator);
+        let stakeholder = creator_info.stakeholder.clone();
+
+        // TODO: make flexible register deposit
+        if let UnregistrationAuthority::Root = unregister_origin {
+            T::Currency::slash_reserved(&stakeholder, T::RegistrationDeposit::get());
+        } else if let UnregistrationAuthority::Creator(who) = unregister_origin {
+            ensure!(who == stakeholder, Error::<T>::NotACreator);
+            T::Currency::unreserve(&stakeholder, T::RegistrationDeposit::get());
+        }
+
+        let current_era = Self::current_era();
+        creator_info.state = CreatorState::Unregistered(current_era);
+        RegisteredCreators::<T>::insert(space_id, creator_info);
+
+        Ok(().into())
+    }
+
+    /// An utility method used to stake specified amount on an arbitrary creator.
+    ///
+    /// `StakerInfo` and `CreatorStakeInfo` are provided and all checks are made to ensure that
+    /// it's possible to complete staking operation.
+    ///
+    /// # Arguments
+    ///
+    /// * `staker_info` - info about staker's stakes on the creator up to current moment
+    /// * `staking_info` - general info about creator stakes up to current moment
+    /// * `value` - value which is being bonded & staked
+    /// * `current_era` - current creators-staking era
+    ///
+    /// # Returns
+    ///
+    /// If stake operation was successful, given structs are properly modified.
+    /// If not, an error is returned and structs are left in an undefined state.
+    pub(super) fn stake_to_creator(
+        staker_info: &mut StakerInfo<BalanceOf<T>>,
+        staking_info: &mut CreatorStakeInfo<BalanceOf<T>>,
+        value: BalanceOf<T>,
+        current_era: EraIndex,
+    ) -> Result<(), Error<T>> {
+        ensure!(
+                !staker_info.latest_staked_value().is_zero() ||
+                    staking_info.number_of_stakers < T::MaxNumberOfStakersPerCreator::get(),
+                Error::<T>::MaxNumberOfStakersExceeded
+            );
+        if staker_info.latest_staked_value().is_zero() {
+            staking_info.number_of_stakers = staking_info.number_of_stakers.saturating_add(1);
+        }
+
+        staker_info
+            .stake(current_era, value)
+            .map_err(|_| Error::<T>::UnexpectedStakeInfoEra)?;
+        ensure!(
+                // One spot should remain for compounding reward claim call
+                staker_info.len() < T::MaxEraStakeValues::get(),
+                Error::<T>::TooManyEraStakeValues
+            );
+        ensure!(
+                staker_info.latest_staked_value() >= T::MinimumStakingAmount::get(),
```

Comment:
Why do we do this check after we actually staked and not before?

Reply:
We want to get the current stake, add the desired stake amount to it (which `stake` function does for us) and then compare whether it's more than the minimum amount.
In other way, we would do the same calculations twice.

Reply:
CONTINUTE FROM HERE xD

Reply:
Add a comment why we do this check after corrections to the stake amount